addons/mail/static/src/js/services/mail_notification_manager.js

```diff
@@ -97,9 +97,10 @@ MailManager.include({
     _handleChannelMessageNotification: function (messageData) {
         var self = this;
         var def;
-        var channelAlreadyInCache = true;
+        var notify = true;
         if (messageData.channel_ids.length === 1) {
-            channelAlreadyInCache = !!this.getChannel(messageData.channel_ids[0]);
+            var channel = this.getChannel(messageData.channel_ids[0]);
+            notify = channel && !(channel.isMassMailing() && session.notification_type == 'email');
```

Comment:
Similar to `notify = channel && (!channel.isMassMailing() || session.notification_type !== 'email');`.

I suppose this diff comes from simplified version of `channel_seen` that happens before sending channel message notification.

I don't really like this, but I don't have a better idea for the moment...
Definitely would put a comment that explains why we do this!